src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+
+tree.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'SPAN') {
+    return;
+  }
+
+  const subTree = e.target.nextElementSibling;
+
+  if (subTree && subTree.tagName === 'UL') {
+    subTree.hidden = !subTree.hidden;
+  }
+});
+
+const addSpansToHeaders = (p) => {
+  for (const li of p.querySelectorAll('li')) {
+    const headerText = li.firstChild;
+
+    if (headerText.nodeType === Node.TEXT_NODE) {
+      const span = document.createElement('span');
+
+      span.textContent = headerText.textContent.trim();
```

Comment:
Trimming the text content is a good practice to remove any leading or trailing whitespace. Ensure that this is the intended behavior for your application.